src/app/room/room.component.ts

```diff
@@ -42,13 +51,14 @@ export class RoomComponent implements OnInit {
     dialogRef.afterClosed().subscribe(name => this.persistPlayer(name));
   }
 
-  // this persistence would probably be better with ngrx
   private persistPlayer(name: string): void {
     this.player = {
       name: name,
-      id: 'random-id',
-      room: this.route.snapshot.params['id']
+      room: this.route.snapshot.params[constants.routeParams.id]
     } as Player;
-    localStorage.setItem(constants.playerKey, JSON.stringify(this.player));
+    this.playerCollection.add(this.player).then(playerReference => {
```

Comment:
Maybe add these firebase calls to a separate service?

Reply:
That is a great idea :)